Apromore-Database/src/main/java/org/apromore/dao/model/DataTypeDao.java

```diff
@@ -0,0 +1,48 @@
+/*-
+ * #%L
+ * This file is part of "Apromore Enterprise Edition".
+ * %%
+ * Copyright (C) 2019 - 2022 Apromore Pty Ltd. All Rights Reserved.
+ * %%
+ * NOTICE:  All information contained herein is, and remains the
+ * property of Apromore Pty Ltd and its suppliers, if any.
+ * The intellectual and technical concepts contained herein are
+ * proprietary to Apromore Pty Ltd and its suppliers and may
+ * be covered by U.S. and Foreign Patents, patents in process,
+ * and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this
+ * material is strictly forbidden unless prior written permission
+ * is obtained from Apromore Pty Ltd.
+ * #L%
+ */
+
+package org.apromore.dao.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.springframework.beans.factory.annotation.Configurable;
+
+@Entity
+@Table(name = "data_type",
+    uniqueConstraints = {
+        @UniqueConstraint(columnNames = {"id"})
+    }
+)
+@Configurable("data_type")
+@Getter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class DataTypeDao extends BaseEntity {
+    @Column(name = "type")
+    @Enumerated(EnumType.STRING)
+    private DataType type;
+}
```

Comment:
If we want to use Builder for creating object, should we include Builder.Default to make it null safe (avoid NPE at runtime)? For example, the createdTimestamp could be null after calling builder.build() without caling builder.createdTimestamp(), and so the created object is left in an unknown state with null createdTimestamp.

Reply:
The createdTimestamp and updatedTimestamp are never manually set. These are taken care of by the respective hibernate annotations.

But your comment is valid for some other attributes (eg. index, attributeTypeDao, dataTypeDao in LogicalLogAttribute). Will check how to add these safety checks